src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1544,11 +1544,11 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
 
     uct_dc_mlx5_update_tx_res(iface, txwq, txqp, pi);
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status, pi, 0);
+    uct_dc_mlx5_iface_set_ep_failed(iface, ep, cqe, txwq, ep_status);
```

Comment:
why needed to move?

Reply:
because have to reset FC window before calling `uct_dc_mlx5_iface_dci_put` which rely on the fact that FC window is updated

Reply:
where does `uct_dc_mlx5_iface_dci_put()` use fc window?

Reply:
https://github.com/openucx/ucx/blob/5f23385e3961c0b1a1f6a307c316ee7aa66041c8/src/uct/ib/dc/dc_mlx5_ep.h#L453
which in turn uses check FC resource
https://github.com/openucx/ucx/blob/5f23385e3961c0b1a1f6a307c316ee7aa66041c8/src/uct/ib/dc/dc_mlx5_ep.h#L370

Reply:
ok, i guess if we released dci earlier, it could be not scheduled to send flush(cancel)?
maybe add short comment?

Reply:
done